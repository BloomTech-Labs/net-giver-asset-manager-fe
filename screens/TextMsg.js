import React, { Component } from "react";
import { Alert, Button, StyleSheet, Text, View } from "react-native";
import Auth0 from "react-native-auth0";
import { AuthSession } from "expo";

const auth0 = new Auth0({
  domain: "dev-dumq2xvs.auth0.com",
  clientId: "cn4Sw78ByHA6bBqVVEihdH3JesIp16o9"
});

class TextMsg extends Component {
  constructor(props) {
    super(props);
    this.state = { accessToken: null };
  }

  onLogin = async () => {
    let result = await AuthSession.startAsync(
      auth0.webAuth
        .authorize({
          scope: "openid profile email"
        })
        .then(credentials => {
          Alert.alert("AccessToken: " + credentials.accessToken);
          this.setState({ accessToken: credentials.accessToken });
        })
        .catch(error => console.log(error))
    );
  };

  onLogout = () => {
    auth0.webAuth
      .clearSession({})
      .then(success => {
        Alert.alert("Logged out!");
        this.setState({ accessToken: null });
      })
      .catch(error => {
        console.log("Log out cancelled");
      });
  };

  render() {
    let loggedIn = this.state.accessToken === null ? false : true;
    return (
      <View style={styles.container}>
        <Text style={styles.header}> NetGiver Auth0Sample - Login </Text>
        <Text>You are{loggedIn ? " " : " not "}logged in . </Text>
        <Button
          onPress={loggedIn ? this.onLogout : this.onLogin}
          title={loggedIn ? "Log Out" : "Log In"}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  header: {
    fontSize: 20,
    textAlign: "center",
    margin: 10
  }
});

export default TextMsg;

// import { AuthSession } from ‘expo’;
// const auth0Domain = ‘https://yourdomain.auth0.com 12’;
// const auth0ClientId = ‘yourid’;

// class TextMsg extends React
