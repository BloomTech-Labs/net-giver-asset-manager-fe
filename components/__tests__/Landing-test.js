import React from 'react';
import renderer from 'react-test-renderer';
import TextMsg from '../../screens/TextMsg';

it(`TextMsg page renders correctly`, () => {
    const tree = renderer.create(<TextMsg />).toJSON();

    expect(tree).toMatchSnapshot();
});
