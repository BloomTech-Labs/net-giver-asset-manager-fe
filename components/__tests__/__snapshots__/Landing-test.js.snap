// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextMsg page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
     NetGiver Auth0Sample - Login 
  </Text>
  <Text>
    You are
     not 
    logged in . 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Log In
      </Text>
    </View>
  </View>
</View>
`;
